Titanium CLI v6.1.0 breaks with X.Y.Z.GA.SOMETHING SDK names

When using Titanium CLI 6.1.0 (this does not occur under previous build 6.0.2), Node v14.20.0 or Node v16.15.1, and doing a distribution build of Titanium 10.1.1.GA, the build breaks with the following exception in sdk.js around a nullpointer when iterating. See error below.

This is caused by the regular expression:

`const versionRegExp = /^(\d+)\.(\d+)\.(\d+)(?:\.\w+)?$/i`

Which won't parse SDKs named like this:

10.1.1.GA.SOMETHING

This is the naming scheme that we use - where SOMETHING indicates certain builds.

If the regular expression is changed to add .* before the $ outside the group:

`const versionRegExp = /^(\d+)\.(\d+)\.(\d+)(?:\.\w+)?.*$/i`

It will ignore other strings at the end of the SDK name.

Can anyone comment? I can put in a PR for this change.

```
using Titanium CLI at location: /usr/local/bin/titanium
Titanium CLI v6.1.0, SDK v10.1.1.GA, https://titaniumsdk.com
Copyright TiDev, Inc. 4/7/2022-Present. All Rights Reserved.

Want to help? https://tidev.io/donate or https://tidev.io/contribute

[ERROR] TypeError: object null is not iterable (cannot read property Symbol(Symbol.iterator))
    at sortVersion (/usr/local/lib/node_modules/titanium/lib/commands/sdk.js:333:45)
    at Array.sort (<anonymous>)
    at Function.select (/usr/local/lib/node_modules/titanium/lib/commands/sdk.js:372:4)
    at Function.selectCallbackified (util.js:293:5)
    at run (/usr/local/lib/node_modules/titanium/lib/commands/sdk.js:75:29)
    at CLI.<anonymous> (/usr/local/lib/node_modules/titanium/lib/cli.js:808:5)
    at wrapper (/usr/local/lib/node_modules/titanium/node_modules/async/dist/async.js:271:20)
    at Object.iteratorCallback (/usr/local/lib/node_modules/titanium/node_modules/async/dist/async.js:501:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/titanium/node_modules/async/dist/async.js:327:20)
    at CLI._fireHookCallback (/usr/local/lib/node_modules/titanium/lib/hook.js:274:12)

error
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,4 +1,8 @@
-6.1.0
+6.1.1 (7/13/2022)
+-------------------
+ * Allow SDK directories to contain any version suffix
+
+6.1.0 (6/28/2022)
 -------------------
  * Re-enabled installing CI builds
  * Updated minor/patch dependencies

--- a/lib/commands/sdk.js
+++ b/lib/commands/sdk.js
@@ -30,7 +30,11 @@ exports.desc = 'manages installed Titanium SDKs';
 /** @namespace SdkSubcommands */
 const SdkSubcommands = {};
 
-const versionRegExp = /^(\d+)\.(\d+)\.(\d+)(?:\.\w+)?$/i;
+// X.Y.Z
+// X.Y.Z.GA or X.Y.Z.RC or X.Y.Z.GA.foo or X.Y.Z.GA.foo.bar
+// X.Y.Z.foo or X.Y.Z.foo.bar
+const versionRegExp = /^(\d+)\.(\d+)\.(\d+)(?:\.\w+)?/i;
+
 const sortTypes = [ 'beta', 'rc', 'ga' ];
 const releaseTypeMap = {
 	latest: 'ga',

--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
 		"mobile web",
 		"appc-client"
 	],
-	"version": "6.1.0",
+	"version": "6.1.1",
 	"author": "TiDev, Inc. <npm@tidev.io>",
 	"bugs": {
 		"url": "https://github.com/tidev/titanium_mobile/issues"

